app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -106,6 +106,14 @@ class ProjectInformationActivity : AppCompatActivity() {
         }
     }
 
+    private fun setFavoriteButtonText(favoriteButton: Button, isFavorite: Boolean?) {
+        favoriteButton.text = when (isFavorite) {
+            null -> LOADING_STRING
+            true -> getString(R.string.favorite_remove_button)
+            false -> getString(R.string.favorite_add_button)
+        }
+    }
+
```

Comment:
Maybe you could disable the button when null  and enable it otherwise, this would avoid undefined behavior when clicking while it is not loaded yet.